rapidsai/legate-boost
# split up PR and nightly/release CI

Contribute sto #115

Splits up CI into 2 different workflows:

* `github-actions` (runs on PRs, does not publish docs)
* `build` (runs on merges to `main` or new tags or via clicking a button in the UI, publishes docs to https://rapidsai.github.io/legate-boost/)

This is a step towards having workflows that automatically publish to the `legate` channel. In such builds, it's not necessary to re-run tests.

## Notes for Reviewers

I'll leave notes inline on the diff explaining the specific design choices here.

### How to test this

When this PR is merged into `main`, we should see a new run of this `build` workflow. In its logs, we should see package builds, docs builds, and docs publishing, but no tests.

The version in the top corner of https://rapidsai.github.io/legate-boost/ should match the version number from the corresponding package build in that same run.

<img width="1494" alt="Screenshot 2024-10-21 at 2 04 04 PM" src="https://github.com/user-attachments/assets/9382ae21-5c1f-4e78-9fbe-3b0e01f0417b">

After that, I'm going to put up another PR that wires up the publishing to the `legate` conda channel. 

### Helpful References

See "Reusing workflows" ([GitHub docs](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow)) for explanation of the syntax.

The structure and naming I'm using is trying to follow RAPIDS conventions... but unlike the non-Legate RAPIDS projects, which get all their workflow definitions from a central public repo (https://github.com/rapidsai/shared-workflows), this is proposing checking those workflow files in directly here.

Maybe in the future, we could have a similar repo of shared workflows for the various RAPIDS `legate-*` projects. Let's defer that for now and just focus on getting `legate-boost` working as an example.
